monai/transforms/intensity/array.py

```diff
@@ -87,7 +89,7 @@ def randomize(self, im_shape: Sequence[int]) -> None:
         super().randomize(None)
         self._noise = self.R.normal(self.mean, self.R.uniform(0, self.std), size=im_shape)
 
-    def __call__(self, img: Union[torch.Tensor, np.ndarray]) -> Union[torch.Tensor, np.ndarray]:
+    def __call__(self, img: DataObjects.Images) -> DataObjects.Images:
```

Comment:
If I use `DataObjects.Images = NdarrayTensor` (which uses `TypeVar`), then I get the following mypy errors:

```
monai/utils/misc.py:359: error: Variable "monai.utils.enums.DataObjects.Images" is not valid as a type  [valid-type]
monai/utils/misc.py:359: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```
where the problematic line of code is:

```python
def convert_data_type(
    data: Any,
    output_type: Optional[type] = None,
    device: Optional[torch.device] = None,
    dtype: Optional[Union[DtypeLike, torch.dtype]] = None,
) -> Tuple[DataObjects.Images, type, Optional[torch.device]]:
```
Any ideas?

Reply:
I replace the DataObjects with NdarrayTensor, and it can pass the mypy check:
```diff
diff --git a/monai/utils/type_conversion.py b/monai/utils/type_conversion.py
index c27fe60e..ac03b881 100644
--- a/monai/utils/type_conversion.py
+++ b/monai/utils/type_conversion.py
@@ -3,8 +3,7 @@ from typing import Any, Optional, Sequence, Tuple, Union
 import numpy as np
 import torch
 
-from monai.config.type_definitions import DtypeLike
-from monai.utils.enums import DataObjects
+from monai.config.type_definitions import DtypeLike, NdarrayTensor
 
 __all__ = [
     "dtype_torch_to_numpy",
@@ -78,7 +77,7 @@ def convert_data_type(
     output_type: Optional[type] = None,
     device: Optional[torch.device] = None,
     dtype: Optional[Union[DtypeLike, torch.dtype]] = None,
-) -> Tuple[DataObjects.Images, type, Optional[torch.device]]:
+) -> Tuple[NdarrayTensor, type, Optional[torch.device]]:
     """
     Convert to `torch.Tensor`/`np.ndarray` from `torch.Tensor`/`np.ndarray`/`float`/`int` etc.
```

Reply:
I also suggest `NdarrayTensor`, thanks.

Reply:
Thanks!